lib/utils/log/slog_text_handler.go

```diff
@@ -35,13 +35,17 @@ import (
 // manner as configured by the Teleport configuration.
 type SlogTextHandler struct {
 	cfg SlogTextHandlerConfig
+	out SlogTextHandlerOutput
 	// withCaller indicates whether the location the log was emitted from
 	// should be included in the output message.
 	withCaller bool
 	// withTimestamp indicates whether the times that the log was emitted at
 	// should be included in the output message.
 	withTimestamp bool
-	// component is the Teleport subcomponent that emitted the log.
+	// rawComponent is the Teleport subcomponent that emitted the log, e.g., "tsh".
+	rawComponent string
```

Comment:
I have to write a test which verifies that this field is properly set in different scenarios and properly copied between clones etc.